ubuntu_image/parser.py

```diff
@@ -320,41 +332,62 @@ def parse(stream_or_string):
             offset_write = structure.get('offset-write')
             size = structure['size']
             structure_type = structure['type']
-            # Validate structure types.  These can be either GUIDs, two hex
-            # digits, hybrids, or the special 'mbr' type.  The basic syntactic
-            # validation happens above in the Voluptuous schema, but here we
-            # need to ensure cross-attribute constraints.  Specifically,
-            # hybrids and 'mbr' are allowed for either schema, but GUID-only
-            # is only allowed for GPT, while 2-digit-only is only allowed for
-            # MBR.  Note too that 2-item tuples are also already ensured.
+            structure_role = structure.get('role')
+            is_mbr = \
```

Comment:
Preferably no backslashes.  I think you can write this as:

```
    is_mbr = (structure_type == 'mbr'
                    or structure_role is StructureRole.mbr)
```

modulo proper formatting that GH's review window apparently doesn't support ;)